### ISSUE TITLE
COCO Eval mAP too low / almost zero || mAP 结果太低 / 几乎为零，请帮忙，谢谢！

### ISSUE THREAD

**Issue by HeZhang33** [meituan/YOLOv6#205]
Dear Meituan Team,

I am encoutering very low / almost zero evaluation mAP results when running COCO128 tiny datase, whereas using the same setttings of YOLOv5 can already give more than 60% mAP.  

I have followed exactly the same steps as you described on installing YOLOv6 for training custom data (e.g. Python 3.8, pip install -r requirements etc.), but still the result does not look normal, compared with the result from YOLOv5 of Ultralytics. 

**My training cmd:** 
python tools/train.py --img 480 --batch 20 --epochs 20 --check-images --check-labels --conf configs/yolov6s.py --data data/coco128.yaml --name exp_test --device 0

**The evaluation result after 20 epochs:**

     Epoch  iou_loss   l1_loss  obj_loss  cls_loss
     19/19     2.832     2.196      5.57     2.707: 100%|██████████| 7/7 [00:05<00:00,  1.20it/s] 
Inferencing model in val datasets.: 100%|███████████████████████████| 4/4 [00:04<00:00,  1.21s/it]

Evaluating speed.

Evaluating mAP by pycocotools.
Saving runs/train/exp_test_yolov612/predictions.json...
loading annotations into memory...
Done (t=0.07s)
creating index...
index created!
Loading and preparing results...
DONE (t=0.27s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.91s).
Accumulating evaluation results...
DONE (t=0.44s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
Epoch: 19 | mAP@0.5: 9.285182502307054e-06 | mAP@0.50:0.95: 2.3144555641754466e-06

Training completed in 0.097 hours.

------------------------------------------------------------------------------------------------------

## You can see that the mAP@0.5 is around 9e-06, far from 0.6 by YOLOv5.

For your reference: the coco128 can be downloaded from https://ultralytics.com/assets/coco128.zip, and the coco128.yaml file that I used is shown as below (train set is used as val here for just quick testing purpose, but still with 20 epochs, the yolov5 already shows mAP0.5 around 0.6 to 0.7 since we are using same 128 images for training and testing of course):

The coco128.yaml file as below:

-------------------------------------------------------------------------------------------

##### train & val & test sets
train: ../datasets/coco128/images/train2017  # train images (relative to 'path') 128 images
val: ../datasets/coco128/images/train2017 # val images (relative to 'path') 128 images
test:  # test images (optional)
#####  number of classes
nc: 80
##### whether it is coco dataset, only coco dataset should be set to True.
is_coco: False
##### class names
names: [ 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light',
         'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
         'elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee',
         'skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard',
         'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
         'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch',
         'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
         'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
         'hair drier', 'toothbrush' ]

-------------------------------------------------------------------------------------------

### Very much appreciate your works and dedication!

**And please help! Many thanks!**

Br, He Zhang

**Comment by mtjhl** [comment]
Hi, when you used yolov5, did you use pertained model?

**Comment by Zephyr69** [comment]
From what I've seen, when training from scratch, yolov6's map grows significantly slower than yolov5's does; you could try to wait longer.

**Comment by HeZhang33** [comment]
yes i was using yolov5s.pt as pretrained model.

He

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: liangliang ***@***.***>
发送时间: Sunday, July 3, 2022 12:28:08 PM
收件人: meituan/YOLOv6 ***@***.***>
抄送: He Zhang ***@***.***>; Author ***@***.***>
主题: Re: [meituan/YOLOv6] COCO Eval mAP too low / almost zero || mAP 结果太低 / 几乎为零，请帮忙，谢谢！ (Issue #205)


Hi, when you used yolov5, did you use pertained model?

―

**Comment by HeZhang33** [comment]
Like I said，i trained yolov5 and yolov6 all using the pretrained s model. It is not waiting longer that simple but should further investigate the reason / bugs behind.



获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Zephyr69 ***@***.***>
发送时间: Sunday, July 3, 2022 4:39:50 PM
收件人: meituan/YOLOv6 ***@***.***>
抄送: He Zhang ***@***.***>; Author ***@***.***>
主题: Re: [meituan/YOLOv6] COCO Eval mAP too low / almost zero || mAP 结果太低 / 几乎为零，请帮忙，谢谢！ (Issue #205)


From what I've seen, when training from scratch, yolov6's map grows significantly slower than yolov5's does; you could try to wait longer.

―

**Comment by HeZhang33** [comment]
OH, I finally find why my results are so low, because I was using "--conf configs/yolov6s.py", which seems to tell yolov6 to start training from scratch, and that is why the mAPs are so low / almost zero in the beginning (naturally as expected). 

When I use "--conf configs/yolov6s_finetune.py", the mAPs all become normal / on-par with the result using yolov5 (e.g. mAP@0.5 gives around 0.75, with 20 epochs on COCO128 mini dataset).

(In yolov5, when using pre-trained weights, one needs to say "--weights yolov5s.pt", whereas in yolov6 here we need to say "--conf configs/yolov6s_finetune.py". This made me mess up the command line args at the first beginning...)

Anyway, thank you Meituan YOLOv6 team. I will further leverage YOLOv6 for more interesting applications in my work in the future.

Happy to close this issue! :)

Br, He
